pydantic/_internal/_known_annotated_metadata.py

```diff
@@ -296,25 +297,28 @@ def _apply_constraint_with_incompatibility_info(
             )
             continue
         elif isinstance(annotation, (at.Predicate, at.Not)):
-            predicate_name = f'{annotation.func.__qualname__}' if hasattr(annotation.func, '__qualname__') else ''
+            predicate_name = f'{annotation.func.__qualname__!r} ' if hasattr(annotation.func, '__qualname__') else ''
 
             def val_func(v: Any) -> Any:
```

Comment:
For performance we could consider formatting the error message before the `def val_func`, so it's reused on each error. Or could have it be cached in some way.

Similarly might want to have the `at.Predicate` / `at.Not` branches define two separate `val_func`?